lab3/euler.cpp

```diff
@@ -0,0 +1,44 @@
+#include <fstream>
+
+int TIMENUMBERS  = 100000;
+double x_0 = 5;
+double v_0 = 0;
+double dt = 0.01;
+double omega = 0.5;
+
+void output(double *x, double *x1, double *v, double *v1, double *t)
+{
+    std::ofstream foute;
+    foute.open("points.txt", std::ios::app);
+    foute << *x << '\t' << *x1 << '\t' << *v << '\t' << *v1 << '\t' << *t << std::endl;
+    foute.close();
+}
+
+
+
+int main()
+{
+    double x = x_0;
+    double x_i;
+    double v = v_0;
+    double x1 = x_0;
+    double x1_i, xz, vz;
+    double v1 = v_0;
+    double t = 0;
+    output(&x, &x1, &v, &v1, &t);
+    for(int i = 0; i < TIMENUMBERS; i++)
+    {
+        x_i = x;
+        x += v*dt;
+        v -= x_i*omega*dt;
+        x1_i = x1;
+        xz = x1 + dt*v1;
+        vz = v1 - omega*x1*dt;
+        x1 += dt*(vz + v1)/2;
+        v1 -= omega*dt*(x1_i + xz)/2;
+        t += dt;
+        if(i%10 == 0){output(&x, &x1, &v, &v1, &t);}
+    }
+    system("python graph.py");
```

Comment:
Not the best solution - to call some python code (potentially unsafe and blocking/throwing exceptions) with system. It's more conventional to call the c programm from some bash script/python wrapper to firstly get the results, and then do post-processing. This way you may separately control the exit codes of your solver and from your post-processing tool. yet still - that's a nice idea to wrap solver and processing up. I'm going to introduce such approaches in a while along with testing procedures.

Reply:
I'll think about it... I always called Python code in this way and nothing ever broke, so I wasn't interested in other ways.